src/sentry/static/sentry/app/actionCreators/group.tsx

```diff
@@ -171,3 +173,199 @@ export function updateNote(
 
   return promise;
 }
+
+type ParamsType = {
+  itemIds?: Array<number> | Array<string>;
+  query?: string;
+  environment?: string | Array<string> | null;
+  project?: Array<number> | null;
+};
+
+type QueryArgs =
+  | {
+      query: string;
+      environment?: string | Array<string>;
+      project?: Array<number>;
+    }
+  | {
+      id: Array<number> | Array<string>;
+      environment?: string | Array<string>;
+      project?: Array<number>;
+    }
+  | {
+      environment?: string | Array<string>;
+      project?: Array<number>;
+    };
+
+/**
+ * Converts input parameters to API-compatible query arguments
+ */
+export function paramsToQueryArgs(params: ParamsType): QueryArgs {
+  const p: QueryArgs = params.itemIds
+    ? {id: params.itemIds} // items matching array of itemids
+    : params.query
+    ? {query: params.query} // items matching search query
+    : {}; // all items
+
+  // only include environment if it is not null/undefined
+  if (params.query && !isNil(params.environment)) {
+    p.environment = params.environment;
+  }
+
+  // only include projects if it is not null/undefined/an empty array
+  if (params.project && params.project.length) {
+    p.project = params.project;
+  }
+
+  // only include date filters if they are not null/undefined
+  if (params.query) {
+    ['start', 'end', 'period', 'utc'].forEach(prop => {
+      if (!isNil(params[prop])) {
+        p[prop === 'period' ? 'statsPeriod' : prop] = params[prop];
+      }
+    });
+  }
+  return p;
+}
+
+function chainUtil<Args extends any[]>(
+  ...funcs: Array<((...args: Args) => any) | undefined>
+) {
+  const filteredFuncs = funcs.filter(
+    (f): f is (...args: Args) => any => typeof f === 'function'
+  );
+  return (...args: Args): void => {
+    filteredFuncs.forEach(func => {
+      func.apply(funcs, args);
+    });
+  };
+}
+
+function wrapRequest(
+  api: Client,
+  path: string,
+  options: RequestOptions,
+  extraParams: RequestCallbacks
+) {
+  if (isUndefined(extraParams)) {
```

Comment:
instead of using lodash, maybe you could just

```suggestion
  if (extraParams !== undefined) {
```
or even better, add `{}` as a default value ?